### DISCUSSION TITLE
The Review ID is not returned by any of the Review endpoints.

### DISCUSSION THREAD

**Discussion by andriymain** [etsy/open-api#1076] [Q&A]
Neither of the endpoints `getReviewsByListing()` or `getReviewsByShop()` contains the Review ID.

**Comment by andriymain** [comment]
```
{
"count": 0,
"results": [
{
"shop_id": 1,
"listing_id": 1,
"rating": 1,
"review": "string",
"language": "string",
"image_url_fullxfull": "string",
"create_timestamp": 946684800,
"created_timestamp": 946684800,
"update_timestamp": 946684800,
"updated_timestamp": 946684800
}
]
}
```

```
{
"count": 0,
"results": [
{
"shop_id": 1,
"listing_id": 1,
"transaction_id": 1,
"buyer_user_id": 1,
"rating": 1,
"review": "",
"language": "string",
"image_url_fullxfull": "string",
"create_timestamp": 946684800,
"created_timestamp": 946684800,
"update_timestamp": 946684800,
"updated_timestamp": 946684800
}
]
}
```

**Reply by Gareth064** [reply]
What do you need the review Id for?

You get the review message for the listing, at least that's what you should be seeing in the "review" field.

There are no other Etsy API endpoints that you can pass the review Id to.

**Reply by andriymain** [reply]
The absence of an ID looks weird, at least it could be used to generate a link:
https://www.etsy.com/shop/{shopName}/reviews/{REVIEW_ID}?ref=SellerDashboard
Moreover, storing the ID in the database would be advantageous.
Also, I would like to save a photo from the review with a reference to review ID

**Reply by suprak** [reply]
Trying to get the reviewer name, but this info is not returned by the API?
